design/deployment_design.md

```diff
@@ -0,0 +1,70 @@
+
+#### Standalone executable
+
+To install the executable without an internet connection, there are a few options that we can pursue. We can give each executable a different name depending on the intended platform. 
+
+```
+conjur.exe (windows)
+conjur (linux)
+conjur-darwin (macOS)
+```
+
+For Windows, executables have the `.exe` in their name so we will follow suit for the CLI Windows executable. Even though the `.exe` is present, the UX remains the same as invoking with just `conjur`.
+
+For Linux and macOS, we will need to differentiate them. For Linux, we can call the executable `conjur` and for macOS, `conjur-darwin`. That way, the common user-case (Linux) will have a cleaner experience. 
+
+A con of this approach would be that on the [release page](https://github.com/cyberark/conjur-api-python3/releases), at face value, the user will not be able to tell which executable is for linux machines. They will need to make an assumption because "linux" is not in the name as with macOS (-darwin) and Window (.exe) executables.
+
+Another option would be pushing zipped folders for each platform to the release page for our Github repository. Each zipped folder will have its respective executable as detailed below.
+
+```
+linux/conjur-cli.zip
+	conjur
+darwin/conjur-cli.zip
+  conjur
+windows/conjur-cli.zip
+  conjur.exe
+```
+
+Once the user has installed the execuable, they can run the executable in one of the following ways:
+
+1. Run the executable like so: `./linux/conjur-cli/conjur --help` 
+2. Move the ***only*** executable (`conjur`) to their `/usr/local/bin` path so that it would look something like `/usr/local/bin/conjur`. That way, the CLI executable will be available from anywhere on their machine without having to detail the full path
+
+TODO: See if part of the installation process we can automatically save executable to the `/usr/local/bin` directory.
+
+*Recommendation*
+
+I propose we pursue the second option of providing a zipped folder and advising the customer to manually move the executable to `/usr/local/bin` path.
+
+#### Upgrade from Ruby to Python CLI
+
+For all customers that installed the Ruby CLI via the gem, we can request that the customer either remove it or rename it. They can do so doing the following:
+
+1. Run `which conjur` and they will get the path of the gem. For example `/Users/myuser/.rvm/gems/ruby-2.x.x/bin/conjur`
+2. Rename the gem found in that path to `/Users/myuser/.rvm/gems/ruby-2.x.x/bin/conjur.v6`
+
+### Testing
+
+TBD
+
+### Open questions
+
+1. When installing the package via pip3, why is it called `conjur-client` instead of `conjur-cli`?
```

Comment:
Because it is not just a cli. It is a library. In python you can consume library by `import`. Which makes sense because I `import conjur_client` in my python program in order to use it (for example: https://github.com/conjurinc/aam-saas-conjur-enterprise/blob/main/scripts/conjur_sanity_check.py). But for every library I can define entry points (see `setup.py`) which will be mapped to `/usr/local/bin/` after installation. Thus, the library nameis conjur client because it provides a Conjur server client wrapper, while there is an additional module in the library which is a cli thus called `conjur-cli`